src/utils.ts

```diff
@@ -1,5 +1,6 @@
 // Copyright 2019 Cognite AS
 
+import { cloneDeepWith } from 'lodash';
```

Comment:
do we actually need a lodash in here?
I mean the library is like 75 kb and we can reduce it by just having a few helper functions
All other places we use it are quite simple
At this place we can just do `JSON.parse(JSON.stringify(obj))` can't we?

Reply:
and to iterate recurcively
```
function eachRecursive(obj)
{
    for (var k in obj)
    {
        if (typeof obj[k] == "object" && obj[k] !== null)
            eachRecursive(obj[k]);
        else
            // check for date
    }
}
```

Reply:
So one thing is to clone, another thing is to replace objects like Date. `JSON.parse(JSON.stringify(obj))` will remove all Date objects with a string which will be harder to work with for us. And when including `cloneDeepWith` from lodash almost all build systems will apply tree shaking to only include what is needed in the final build